my_lru_cache.py

```diff
@@ -0,0 +1,170 @@
+            input_args_kwargs = prepare_input_data(*args, **kwargs)
+            if func_key_str in cache:  # Checking result of the function is available with the given parameters in the cache
+                for unit in cache[func_key_str]:
+                    if input_args_kwargs in unit["input_args"]:
+                        cache[f'cache_usage_{func_key_str}'] += 1
+                        unit["time"] = str(datetime.now())  # changing last used time
+                        json_dump(cache)
+                        return unit["func_do_values"]
+            result_func = func(*args, **kwargs)
+            clear_cache_on_overflow(cache)  # Checking if the cache is full
+            new_unit_in_cache = {"input_args": input_args_kwargs,  # Create a new unit in cache
+                                 "func_do_values": result_func,
+                                 "time": str(datetime.now())}
+            if not func_key_str in cache:
+                cache[f"cache_usage_{func_key_str}"] = 0
+                cache[f"usage_{func_key_str}"] = 0
+                cache[func_key_str] = []
+            cache[func_key_str].append(new_unit_in_cache)
+            cache[f"usage_{func_key_str}"] += 1
+            json_dump(cache)
+            return result_func
+
+        def cache_info():
+            """Return information about used cache, used caching function and free space in cache"""
+            try:
+                free_cache_space = max_size - all_space_cache()
+                calculated_func = cache[f"usage_{func_key_str}"]
+                cache_usage = cache[f"cache_usage_{func_key_str}"]
+                return (f"Function {func_key_str} is calculated {calculated_func}\nValue is taken from the"
+                        f" cache {cache_usage}\nFree space in cache {free_cache_space}")
+            except KeyError:
+                return "Сache not found"
+
+        def cache_clear():
+            """Clear cache file"""
+            try:
+                path = os.path.abspath(f'cache/cache_{func_key_str}.json')
+                os.remove(path)
+                print("Cache is cleared")
+            except FileNotFoundError:
+                print("Сache is already cleared")
+
+        def json_dump(cache):
+            try:
+                with open(os.path.join(os.path.abspath(f'cache/cache_{func_key_str}.json')), 'w',
+                          encoding='UTF-8') as file:
+                    json.dump(cache, file, indent=2)
+            except FileNotFoundError:
+                os.mkdir('cache')
+                with open(os.path.join(os.path.abspath(f'cache/cache_{func_key_str}.json')), 'w',
+                          encoding='UTF-8') as file:
+                    json.dump(cache, file, indent=2)
+
+        def json_load():
+            try:
+                with open(f'cache/cache_{func_key_str}.json', 'r', encoding='UTF-8') as file:
+                    return json.load(file)
+            except FileNotFoundError:
```

Comment:
here you absolutely right when checking if a file exists, because the mode is' r'